cmd/controller-bootstrap/command/generate.go

```diff
@@ -14,19 +14,132 @@
 package command
 
 import (
+	"bytes"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"text/template"
+
 	"github.com/spf13/cobra"
 )
 
+type templateVars struct {
+	*metaVars
+	AWSSDKGoVersion    string
+	RuntimeVersion     string
+	TestInfraCommitSHA string
+}
+
 var templateCmd = &cobra.Command{
 	Use:   "generate",
 	Short: "generate template files in an ACK service controller repository",
 	RunE:  generateController,
 }
 
-// TODO: generateController creates the initial directories and files for a service controller
+// generateController creates the initial directories and files for a service controller
 // repository by rendering go template files.
-// When a controller is already existing, then this method only updates the project
+// TODO: When a controller is already existing, then this method only updates the project
 // description files.
 func generateController(cmd *cobra.Command, args []string) error {
+	if err := validateRequiredOptions(); err != nil {
```

Comment:
validation does not need to be only for required arguments.
```suggestion
	if err := validateArgs(); err != nil {
```